model/src/main/java/org/jboss/pnc/model/RepositoryConfiguration.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.model;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.SequenceGenerator;
+import javax.validation.constraints.Size;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The JPA entity class that contains configuration of the SCM repositories.
+ *
+ * @author Jakub Bartecek
+ */
+@Entity
+public class RepositoryConfiguration implements GenericEntity<Integer> {
+
+    private static final long serialVersionUID = 4248038054068607536L;
+
+    private static final String SEQ_NAME = "repository_configuration_id_seq";
+
+    @Id
+    @SequenceGenerator(name = SEQ_NAME, sequenceName = SEQ_NAME, allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = SEQ_NAME)
+    private Integer id;
+
+    /**
+     * Repository URL containing project to be build.
+     * This URL *MUST* be read/write.
+     */
+    @Size(max = 255)
+    @Column(unique = true, nullable = false, updatable = false)
```

Comment:
Yes, I agree. It's not the best.

So, what about:
/**
URL to the internal Git repository, which is the main repository used for the builds. New commits are being added to this repository, during the builds
*/
internal

/**
URL to the upstream repository.
*/
external